### ISSUE TITLE
ValueError: Dont know how to convert flask.ext.admin.form.rules.FieldSet object

### ISSUE THREAD

**Issue by ye** [flask-admin/flask-admin#1079]
I've recently updated the Flask Admin package from `1.0.9` to `1.3.0` and the working existing code stopped working with Error:

```python
Traceback (most recent call last):
  File "/api/admin/app.py", line 1188, in <module>
    category='Practitioner'))
  File "/usr/local/lib/python3.4/site-packages/flask_admin/contrib/sqla/view.py", line 291, in __init__
    menu_icon_value=menu_icon_value)
  File "/usr/local/lib/python3.4/site-packages/flask_admin/model/base.py", line 633, in __init__
    self._refresh_cache()
  File "/usr/local/lib/python3.4/site-packages/flask_admin/model/base.py", line 746, in _refresh_cache
    self._refresh_form_rules_cache()
  File "/usr/local/lib/python3.4/site-packages/flask_admin/model/base.py", line 696, in _refresh_form_rules_cache
    form_rules = rules.RuleSet(self, self.form_rules)
  File "/usr/local/lib/python3.4/site-packages/flask_admin/form/rules.py", line 372, in __init__
    self.rules = self.configure_rules(rules)
  File "/usr/local/lib/python3.4/site-packages/flask_admin/form/rules.py", line 408, in configure_rules
    raise ValueError('Dont know how to convert %s' % repr(r))
ValueError: Dont know how to convert <flask.ext.admin.form.rules.FieldSet object at 0x7fae6fca2e10>
```

The offending code line looks like this:
![screen shot 2015-10-12 at 2 14 45 pm](https://cloud.githubusercontent.com/assets/133924/10435110/fdf04b50-70eb-11e5-961d-415a08d85c89.png)

 * Python is `3.4.1`
 * SQLAlchemy is `0.9.8`
 * Flask is `0.10.1`

Anyone knows why?

**Comment by mrjoes** [comment]
Has something to do with your rendering rules configuration. Can you paste appropriate snippet?

**Comment by ye** [comment]
@mrjoes here you go. The code has been working with Flask Admin `1.0.9` though.

```python
class PractitionerProfileView(LoggedInView):
    form_rules = [
        rules.FieldSet(('user', 'role'),
                       'User'),
        rules.FieldSet(('phone_number', 'stripe_id'
                        'show_when_unavailable', ),
                       'Profile'),
        rules.FieldSet(('categories', 'certifications', 'specialties',
						'default_cancellation_policy', 'booking_buffer'),
                       'Professional'),
    ]
```

**Comment by mrjoes** [comment]
Your code looks fine. Are you sure you don't have two version of Flask-Admin installed and `rules.FieldSet` is imported from old version?

**Comment by ye** [comment]
`from flask.ext.admin.form import rules` is the only way `rules` is being imported.

**Comment by mrjoes** [comment]
Do you import view same way?

I suspect that you have two versions of Flask-Admin installed and view is from one version and `FieldSet` is from another.

**Comment by ye** [comment]
Well, all the admin code are within one file  `admin/app.py` so yeah, I don't really import external views. And I doubt I have two version of Flask-Admin as everything is containerized with very lean pip requirements. Here are an excerpt with the relevant pip packages.

```bash
root@f612da4931b7:/api/admin# pip freeze
Flask==0.10.1
Flask-Admin==1.0.9
Flask-SQLAlchemy==2.0
Jinja2==2.7.3
MarkupSafe==0.23
PyJWT==0.3.1
PyMySQL==0.6.3
PySocks==1.5.6
SQLAlchemy==0.9.8
WTForms==2.0.1
Werkzeug==0.9.6
...
```

**Comment by mrjoes** [comment]
But you mentioned you use Flask-Admin 1.3.0, why dod you still have Flask-Admin 1.0.9 in your package list?

**Comment by ye** [comment]
Sorry, that was a wrong docker image. Here is the right one.

```bash
$ docker-compose run --rm admin pip freeze
Flask==0.10.1
Flask-Admin==1.3.0
Flask-SQLAlchemy==2.0
Jinja2==2.7.3
MarkupSafe==0.23
PyJWT==0.3.1
PyMySQL==0.6.3
PySocks==1.5.6
SQLAlchemy==0.9.8
WTForms==2.0.1
Werkzeug==0.9.6
...
```

**Comment by mrjoes** [comment]
No ideas. Can check import paths for your `ModelView` and `FieldSet`?

Just in case here are lines in question: https://github.com/flask-admin/flask-admin/blob/master/flask_admin/form/rules.py#L403

And here's FieldSet (it is child of BaseRule): https://github.com/flask-admin/flask-admin/blob/master/flask_admin/form/rules.py#L335

**Comment by ye** [comment]
`from flask.ext.admin.contrib.sqla import ModelView` is how I import `ModelView`.  As far as `FieldSet` goes, it's `rules.Fieldset` and `from flask.ext.admin.form import rules`.

And yeah, I am aware of those lines are emitting the errors. I will try to print out the context around those lines to get a further look. Thanks.

**Comment by pawl** [comment]
This might be related to this issue?: https://github.com/flask-admin/flask-admin/issues/917

**Comment by ye** [comment]
@pawl That's it (for the most part)! Thank you very much! 

However, while I was able to get passed the ValueError named in the ticket, I encountered a bunch of Warnings like this:

```bash
/usr/local/lib/python3.4/site-packages/flask_admin/model/base.py:1139: UserWarning: Fields missing from ruleset: categories
  warnings.warn(text)
```

AFAIK, they are most likely emitted via 
https://github.com/flask-admin/flask-admin/blob/master/flask_admin/model/base.py#L1236

Again, what's been changed for visible_fields vs. missing_fields from `1.0.9` to `1.3.0`? I am confused because the code was running pristine with zero warning / error...
